### PR TITLE
drivers: gpio_nrfx: Fix handling of GPIO_DISCONNECTED

### LINKED ISSUE #47030: drivers: gpio: nrfx: return -ENOTSUP rather than -EIO for misconfigurations
**Describe the bug**
The `gpio_nrfx.c` driver does not support configuring with `GPIO_DISCONNECTED`, but rather than report `-ENOTSUP`, it reports `-EIO`, which breaks the GPIO driver API.

According to the API, `-EIO` is reserved for "I/O error when accessing an external GPIO chip", whereas `-ENOTSUP` is used for "if any of the configuration options is not supported".

**To Reproduce**
1. west build -b nrf52840dk_nrf52840 -t flash tests/drivers/gpio/gpio_get_direction
2. See error

**Expected behavior**
The driver should report `-ENOTSUP` and the tests with unsupported configuration should be skipped.
See https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/gpio.h#L675

The solution to this should also create additional tests in `tests/drivers/gpio/gpio_basic_api/` to test for a valid return value when setting `GPIO_INPUT | GPIO_OUTPUT | GPIO_OUTPUT_INIT_LOW` or `GPIO_DISCONNECTED`, as those settings seem to be unsupported on some platforms.

**Impact**
GPIO driver tests should fail if they do not follow the API properly.

**Logs and console output**
```
DEBUG   - DEVICE: START - test_disconnect
DEBUG   - DEVICE: 
DEBUG   - DEVICE: Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/gpio/gpio_get_direction/src/main.c:20: test_disconnect: (0 not equal to rv)
DEBUG   - DEVICE: gpio_pin_configure() failed: -5
DEBUG   - DEVICE: FAIL - test_disconnect in 0.15 seconds
```

**Environment (please complete the following information):**
 - OS: ?
 - Toolchain ?
 - Commit SHA or Version used: 3f0e8dda3d72bd9dda069bc0c013d7d67f593c7e

**Additional context**
See
https://github.com/zephyrproject-rtos/zephyr/issues/46917#issuecomment-1168477276
https://github.com/zephyrproject-rtos/zephyr/issues/46917#issuecomment-1169877231

### PR DESCRIPTION
**Pull request by anangl**
The driver returns -EIO when a pin to be disconnected was not earlier
configured as input or output, what is not in line with the GPIO API.
This commit changes the driver to return 0 in such case.
Also -EIO is incorrectly returned when an interrupt trigger cannot be
configured in the nrfx_gpiote driver. This commit corrects this value
to -EINVAL.

Fixes #47030.

### CODE REVIEW

**Review by henrikbrixandersen — DISMISSED** [review]
LGTM

**Comment by cfriedt** [comment]
I wonder if it would be wise to wait until after #46960 is merged. @anangl - did you have a chance to test it?

**Comment by anangl** [comment]
> I wonder if it would be wise to wait until after #46960 is merged. @anangl - did you have a chance to test it?

@cfriedt I've tested it now. The `test_input_output` case fails, but that's because of: https://github.com/zephyrproject-rtos/zephyr/pull/46960#discussion_r910643219.

**Review by nordic-krch — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**
